configs/machines/aleph.yaml

```diff
@@ -46,53 +46,127 @@ hyper_flag: ""
+                        NETCDFROOT: $IO_LIB_ROOT
+                        NETCDFFROOT: $IO_LIB_ROOT
+                        ECCODESROOT: $IO_LIB_ROOT
+
+                        HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
+                        NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
+                        NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+                        NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+                        OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
+                        PATH: $IO_LIB_ROOT/bin:$PATH
+                        MPIFC: ftn
+
+        intel_mpich:
+                fc: ftn
+                f77: ftn
+                mpifc: ftn
+                mpicc: cc
+                cc: cc
+                cxx: CC
+                module_actions:
+                        - "unload PrgEnv-cray"
+                        - "load PrgEnv-intel/6.0.5"
+                        - "load craype-x86-skylake"
+                        - "load cray-mpich/7.7.3"
+                        - "load gcc/7.3.0"
+                        - "load pbs"
+                        - "load cmake/3.14.0"
+                        - "load cdo/1.9.5"
+                        - "load fftw/2.1.5.9"
+                        - "load nco/4.9.4"
+                        - "load proj4/5.1.0"
+                        - "load python/3.9.1"
+                        - "load cray-netcdf-hdf5parallel/4.6.1.3"
+                        - "load cray-hdf5-parallel/1.10.2.0"
+                        - "list"
+                add_export_vars:
+                        ECCODESROOT: /home/awiiccp2/software/HPC_libraries/eccodes_intel_mpich/
+
+                        HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
+                        NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
+                        NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+                        NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+                        OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
+                        PATH: $IO_LIB_ROOT/bin:$PATH
+                        MPIFC: ftn
 export_vars:
         LC_ALL: en_US.UTF-8
 
-        HDF5ROOT: $HDF5_ROOT
-
-        NETCDFFROOT: $NETCDF_DIR
-        NETCDFROOT: $NETCDF_DIR
-        NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-        NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-        NETCDF_CXX_LIBRARIES: $NETCDFROOT/lib
-
         PERL5LIB: /usr/lib64/perl5
-        #LAPACK_LIB: '"-lmkl_intel_lp64 -lmkl_core -mkl=sequential -lpthread -lm -ldl"'
-        #LAPACK_LIB_DEFAULT: '"-L/global/AWIsoft/intel/2018/compilers_and_libraries_2018.5.274/linux/mkl/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_sequential"'
         XML2ROOT: /usr
-        ZLIBROOT: /usr
         TMPDIR: /tmp
         # Prepend labels to the stdout/sterr lines
         PMI_LABEL_ERROUT: 1
 
-        ECCODESROOT: /proj/awiiccp/software/ecmwf/eccodes_cce_mpich
```

Comment:
You don't need to remove this since a while now. This variables in `export_vars` are place holders and get substituted by whatever you define in `add_export_vars`. This should avoid repetition in the `add_export_vars` within the choose blocks.

Reply:
I see. I would say any developer that adds a new compiler and mpi environment should make a conscious decision about where these come from. Do I get them from my own library installation or from a module? What is the path called in the module?
The defaults invite developers not to think about these questions. Instinctively I would not use defaults for these things.